From 53f7d70efc43667d35b9ac0a58531e1878217f61 Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Tue, 9 Jan 2018 15:41:02 +0000
Subject: [PATCH 26/27] DevelopmentSettings: Hide OEM unlock by default

 * The "OEM unlocking" toogle should not be visible on any
   device running LineageOS, for obvious reasons. Allow it only
   to show up for Google devices, such as Pixel, that set
   ro.oem_unlock_supported=1.

Change-Id: I2b09974fc24e783b985de213b504e964771cbcb7
---
 .../settings/development/OemUnlockPreferenceController.java    | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/com/android/settings/development/OemUnlockPreferenceController.java b/src/com/android/settings/development/OemUnlockPreferenceController.java
index 95fe9b520a..65eaca2fe0 100644
--- a/src/com/android/settings/development/OemUnlockPreferenceController.java
+++ b/src/com/android/settings/development/OemUnlockPreferenceController.java
@@ -23,6 +23,7 @@ import android.app.Activity;
 import android.content.Context;
 import android.content.Intent;
 import android.content.res.Resources;
+import android.os.SystemProperties;
 import android.os.UserHandle;
 import android.os.UserManager;
 import android.service.oemlock.OemLockManager;
@@ -65,7 +66,7 @@ public class OemUnlockPreferenceController extends DeveloperOptionsPreferenceCon
 
     @Override
     public boolean isAvailable() {
-        return mOemLockManager != null;
+        return mOemLockManager != null && SystemProperties.getBoolean("ro.oem_unlock_supported", false);
     }
 
     @Override
-- 
2.17.1

